### PR TITLE
Not unscaling the predictions on the validation set

### PR DESCRIPTION
**Pull request by shihchengli**
## Description
The predictions on the validation set are unscaled, while the targets are scaled. Therefore, the `val_loss` is incorrect. I can see that on [line 170](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/models/model.py#L170) in model.py, `preds = self(bmg, V_d, X_d)` would return an unscaled value, and on [line 145](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/models/model.py#L145), `preds = self.predictor.train_step(Z)` would return a scaled value. At first glance, those two functions should be the same, but the `output_transform` does a trick that can only not be used during training (see [here](https://github.com/hwpang/chemprop/blob/9c51462c8d5c03852384b6ab6e5f53ef75a9d657/chemprop/nn/transforms.py#L37-L42)). Therefore, we cannot just call `train_step` in the `validation_step` to solve this problem. I thus added the `train_step` back to predictors, which is removed in PR #726 and modified the code in `_evaluate_batch` to not unscale the predictions during validation.

### CODE REVIEW

**Review by JacksonBurns — CHANGES_REQUESTED** [review]
Can you try doing this with the lightning attributes instead, i.e. `self.trainer.state.fn`: https://lightning.ai/docs/pytorch/stable/common/trainer.html#state

Want to avoid duplicating their code.

Thanks!

**Comment by davidegraff** [comment]
You can also just call `output_transform.train()` in the `on_begin_validation_epoch` hook. A bit hacky, but we can’t get around this if we want our model to scale test outputs yet also want comparable training and validation outputs

**Comment by shihchengli** [comment]
Thanks! I like this idea. I agree that it's a bit hacky, but it seems more confusing to compare the two blocks below:
1. in `training_step`:
```
Z = self.fingerprint(bmg, V_d, X_d)
preds = self.predictor.train_step(Z)
```
2. in `_evaluate_batch`:
```
preds = self(bmg, V_d, X_d)
```
They are the same because `forward()` in the MPNN class is defined as `self.predictor(self.fingerprint(bmg, V_d, X_d))`, and both `self.predictor.train_step` and `self.predictor.forward` are defined the same way in `RegressionFFN`. After further investigation, I can finally understand that the `UnscaleTransform` object only scales the predictions in the training set. I don't know if this is documented anywhere. It might be quite confusing for users playing around with these modules.

**Comment by davidegraff** [comment]
With the addition of the new `output_transform` attribute, there's no need to separately define `train_step()` and `forward()` anymore. For now, this works, but a future PR should remove this method and use one of the suggestions here to not scale outputs during validation.

**Review by hwpang — APPROVED** [review]
LGTM. Perhaps we should use the `*Transform` classes in the integration tests to catch this type of error. That can be a separate PR.

**Comment by davidegraff** [comment]
> LGTM. Perhaps we should use the `*Transform` classes in the integration tests to catch this type of error. That can be a separate PR.

good point! We should have:
- unit tests for our transforms: ensure the math/implementation is correct
- integration tests for our new models: ensure that training and evaluation output domains are different, i.e., our training outputs should have $\hat \mu \simeq 0, \hat \sigma \simeq 1$ and our evaluation outputs should have $\hat \mu \simeq \mu, \hat \sigma \simeq \sigma$ (roughly equal to our training statistics)

**Review by JacksonBurns — APPROVED** [review]
:+1:
